x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -338,6 +340,18 @@ public Expression visitLogicalIn(EsqlBaseParser.LogicalInContext ctx) {
         return ctx.NOT() == null ? e : new Not(source, e);
     }
 
+    @Override
+    public Object visitIsNull(EsqlBaseParser.IsNullContext ctx) {
+        Expression exp = expression(ctx.valueExpression());
+        Source source = source(ctx.valueExpression(), ctx);
+
+        if (ctx.NOT() != null) {
+            return new IsNotNull(source, exp);
+        } else {
+            return new IsNull(source, exp);
+        }
+    }
```

Comment:
Use the ternary operator to make it a one-liner:
`return ctx. NOT() != null ? new IsNotNull(source, exp) : new IsNull(source, exp);`